tests/conftest.py

```diff
@@ -3,38 +3,45 @@
 from unittest.mock import patch
 from sys import modules
 
+PDIR2_CONFIG_FILE = "PDIR2_CONFIG_FILE"
+
 def remove_module_cache():
-    for module in (
-        'pdir',
-        'pdir.api',
-        'pdir.constants',
-        'pdir.format',
-        'pdir.configuration',
-        'pdir.color',
-        'pdir.utils',
-    ):
+    """
+    There are some global settings which are initialized when pdir firstly
+    import, then after that, no matter how you change the config or mock
+    ``isatty()`` functions, those global values (settings, mostly), will
+    not change.
+
+    So in order to make some of our patches or test configurations work, we
+    need to clear the import cache. So that when we ``import pdir`` in test
+    cases, those global values will be initialized again due to the lack of
+    cache.
```

Comment:
I think it would be helpful to add a link to your blog post, even if it's Chinese, people can use a translator. Again, thanks for writing it!